rest/config.go

```diff
@@ -676,6 +677,34 @@ func (config *ServerConfig) setupAndValidateDatabases() (errs error) {
 	return nil
 }
 
+// expandEnv replaces $var or ${var} in config according to the values of the
+// current environment variables. The replacement is case-sensitive. References
+// to undefined variables are replaced by an empty string. A default value can
+// be given by using the form ${var:-default value}.
+func expandEnv(config []byte) []byte {
+	return []byte(os.Expand(string(config), func(key string) string {
+		if key == "$" {
+			base.Debugf(base.KeyAll, "Skipping environment variable expansion: %s", key)
+			return key
+		}
+		kvPair := strings.SplitN(key, ":-", 2)
+		key = kvPair[0]
+
+		value := os.Getenv(key)
+		if value == "" && len(kvPair) == 2 {
+			// Set value to the default.
+			value = kvPair[1]
+			base.Debugf(base.KeyAll, "Replacing config environment variable '${%s}' with "+
+				"default '%s'", key, base.UD(kvPair[1]))
+		} else {
+			base.Debugf(base.KeyAll, "Replacing config environment variable '${%s}' with '%s'",
+				key, base.UD(value))
+		}
+
+		return value
+	}))
+}
```

Comment:
I'd suggest pulling this default parameter expansion out into another function, it's fairly standalone, and easier to test if the test only needs to worry about a single key.

An added benefit is that you can pass through the function to get an env var value so that you don't actually need to set/unset any in the test itself.

```suggestion
		return envDefaultExpansion(key)
	}))
}

// Implements the ${foo:-bar} parameter expansion from
// https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02
func envDefaultExpansion(key string) (value string) {
	kvPair := strings.SplitN(key, ":-", 2)
	key = kvPair[0]

	value = os.Getenv(key)
	if value == "" && len(kvPair) == 2 {
		// Set value to the default.
		value = kvPair[1]
		base.Debugf(base.KeyAll, "Replacing config environment variable '${%s}' with "+
			"default '%s'", key, base.UD(value))
	} else {
		base.Debugf(base.KeyAll, "Replacing config environment variable '${%s}' with '%s'",
			key, base.UD(value))
	}

	return value
}
```

Reply:
Sure, added changes to encapsulate the inline function inside envDefaultExpansion to make testing the default parameter expansion easier